src/zenml/stack/utils.py

```diff
@@ -0,0 +1,152 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Util functions for handling stacks, components, and flavors."""
+
+from typing import Any, Dict, Optional
+
+from zenml.client import Client
+from zenml.enums import StackComponentType, StoreType
+from zenml.logger import get_logger
+from zenml.models.flavor_models import FlavorFilterModel, FlavorResponseModel
+from zenml.stack.flavor import Flavor
+from zenml.stack.stack_component import StackComponentConfig
+from zenml.zen_stores.base_zen_store import BaseZenStore
+
+logger = get_logger(__name__)
+
+
+def validate_stack_component_config(
+    configuration_dict: Dict[str, Any],
+    flavor_name: str,
+    component_type: StackComponentType,
+    zen_store: Optional[BaseZenStore] = None,
+) -> StackComponentConfig:
+    """Validate the configuration of a stack component.
+
+    Args:
+        configuration_dict: The stack component configuration to validate.
+        flavor_name: The name of the flavor of the stack component.
+        component_type: The type of the stack component.
+        zen_store: An optional ZenStore in which to look for the flavor. If not
+            provided, the flavor will be fetched via the regular ZenML Client.
+            This is mainly useful for checks running inside the ZenML server.
+
+    Returns:
+        The validated stack component configuration.
+
+    Raises:
+        ValueError: If the configuration is invalid.
+    """
+    flavor_class = get_stack_component_flavor_class(
+        flavor_name=flavor_name,
+        component_type=component_type,
+        zen_store=zen_store,
+    )
+    configuration = flavor_class.config_class(**configuration_dict)
+    if not configuration.is_valid:
+        raise ValueError(
+            f"Invalid stack component configuration. Please verify "
+            f"the configurations set for {component_type}."
+        )
+    if not zen_store:  # Doesn't make sense to warn inside a server.
```

Comment:
Instead of having this unclear `if` condition here, I suggest we don't call this warning here, but instead simply call the `_warn_if_config_server_mismatch` function from the client after the configuration was validated

Reply:
Yeah feels cleaner, agreed 👍